proof/refine/ARM/IpcCancel_R.thy

```diff
@@ -2849,12 +2992,14 @@ lemma cancelAllIPC_invs'[wp]:
    apply (metis fold_list_refs_of_replies')
   apply (clarsimp simp: sym_refs_asrt_def)
   apply (rule conjI)
-   apply (clarsimp simp: valid_ep'_def split: endpoint.splits)
-   apply (clarsimp simp: valid_ep'_def)
    apply (drule(1) sym_refs_ko_atD')
-   apply (clarsimp simp: st_tcb_at_refs_of_rev' split: endpoint.splits)
-    apply (clarsimp elim!: if_live_state_refsE split: if_splits
-           | drule(1) bspec | drule st_tcb_at_state_refs_ofD')+
+   apply (clarsimp simp: valid_ep'_def st_tcb_at_refs_of_rev' split: endpoint.splits)
+    apply ((intro conjI,
+            ((drule(1) bspec | drule st_tcb_at_state_refs_ofD'
+              | clarsimp elim!: if_live_state_refsE split: if_splits)+)[1],
+            fastforce simp: runnable'_def st_tcb_at'_def obj_at'_def,
+            fastforce elim!: pred_tcb'_weakenE)+)[2]
```

Comment:
I do like combining proofs into one apply but this might be a bit much. I don't see this being easy to maintain if it ever needs to change.

I think the best alternative here would be if you manage to do it with a single auto or fastforce. If that's not possible, how big is it when you do it step by step?